engine/table/src/main/java/io/deephaven/engine/table/impl/partitioned/PartitionedTableImpl.java

```diff
@@ -318,6 +319,26 @@ public PartitionedTable transform(
         return resultPartitionedTable;
     }
 
+    /**
+     * Ensures that the returned executionContext will have an OperationInitializer compatible with being called by work
+     * already running on an initialization thread - it must either already return false for
+     * {@link OperationInitializer#canParallelize()}, or must be a different instance than the current context's
+     * OperationInitializer.
+     */
+    private static ExecutionContext maybeReplaceExecContext(ExecutionContext provided) {
+        if (provided == null) {
+            return null;
+        }
+        ExecutionContext current = ExecutionContext.getContext();
+        if (!provided.getInitializer().canParallelize()) {
+            return provided;
+        }
+        if (current.getInitializer() != provided.getInitializer()) {
+            return provided;
+        }
+        return provided.withOperationInitializer(OperationInitializer.NON_PARALLELIZABLE);
```

Comment:
I am finding the name and javadoc a little confusing.

I'd like to suggest renaming to `disableRecursiveParallelOperationInitialization()` with the javadoc indicating that it returns the provided execution context after possibly disabling parallelism if further parallelism could result in deadlock.

Then probably finish off the method with the comment making it clear this is a disable:
```java
// disable recursive parallelism
return provided.withOperationInitializer(...);
```